eureka2-read-server/src/main/java/com/netflix/eureka2/server/interest/FullFetchInterestClient.java

```diff
@@ -0,0 +1,70 @@
+package com.netflix.eureka2.server.interest;
+
+import javax.inject.Inject;
+
+import com.netflix.eureka2.channel.ChannelFactory;
+import com.netflix.eureka2.channel.InterestChannel;
+import com.netflix.eureka2.client.interest.AbstractInterestClient;
+import com.netflix.eureka2.client.interest.EurekaInterestClient;
+import com.netflix.eureka2.connection.RetryableConnection;
+import com.netflix.eureka2.connection.RetryableConnectionFactory;
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.interests.Interests;
+import com.netflix.eureka2.registry.SourcedEurekaRegistry;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import rx.Observable;
+import rx.functions.Func1;
+
+/**
+ * {@link EurekaInterestClient} implementation with single full registry fetch subscription.
+ * Client interest subscriptions are not propagated to the channel, as the registry is already eagerly
+ * subscribed to the full content.
+ *
+ * @author Tomasz Bak
+ */
+public class FullFetchInterestClient extends AbstractInterestClient {
```

Comment:
Seems like the standard InterestClient should be able to be used for the same purpose as this special client, as long as we do an eager forInterest(FullRegistryInterest) from the ReadRegistry?